app/assets/javascripts/application.js

```diff
@@ -0,0 +1,34 @@
+'use strict'
+
+document.onreadystatechange = function () {
+  var csrf = document.querySelector('meta[name="csrf-token"]').content
+  if (document.readyState === 'interactive') {
+    var basePath = document.getElementById('base-path-id')
+    var documentTitle = document.getElementById('document-title-id')
+    var pathArray = document.location.pathname.split('/')
+    var documentId = pathArray[2]
+    var url = new URL(document.location.origin + '/documents/' + documentId + '/generate-path')
+    documentTitle.onblur = function () {
+      if (!documentTitle.value) {
+        basePath.innerHTML = "You haven’t entered a title yet."
+        return
+      }
+      url.searchParams.append('title', documentTitle.value)
+      window.fetch(url, {
+        credentials: 'include',
+        headers: {
+          'Content-Type': 'application/json; charset=utf-8',
+          'X-CSRF-Token': csrf
+        },
+      }).then(function(response) {
```

Comment:
Would be good to check the response status and examine the response only in case of 200
```
     }).then(
        function (response) {
          if (response.status !== 200) {
            console.error('Looks like there was a problem. Status Code: ' +
              response.status)
            return
          }
          // Examine the response
          response.json().then(function (result) {
            var basePath = document.getElementById('base-path-id')
            if (result.reserved) {
              basePath.innerHTML = 'www.gov.uk' + result['base_path']
            } else {
              basePath.innerHTML = 'Path is taken, please edit the title.'
            }
          })
        }).catch(function (err) {
        console.error(err)
      })
```